src/webob/acceptparse.py

```diff
@@ -268,6 +667,1250 @@ def _match(self, mask, item):
+           :meth:`AcceptLanguageValidHeader.quality`).
+
+        :param offer: (``str``) language tag offer
+        :return: (``float``) ``1.0``.
+
+        When the ``Accept-Language`` header is invalid or not in the request,
+        all offers are equally acceptable, so 1.0 is always returned.
+        """
+        warnings.warn(
+            'The behavior of .quality for the AcceptLanguage classes is '
+            'currently being maintained for backward compatibility, but the '
+            'method may be deprecated in the future, as its behavior is not '
+            'specified in (and currently does not conform to) RFC 7231.',
+            PendingDeprecationWarning,
+        )
+        return 1.0
+
+
+class AcceptLanguageNoHeader(_AcceptLanguageInvalidOrNoHeader):
+    """
+    Represent when there is no ``Accept-Language`` header in the request.
+
+    This object should not be modified. To add to the header, we can use the
+    addition operators (``+`` and ``+=``), which return a new object (see the
+    docstring for :meth:`AcceptLanguageNoHeader.__add__`).
+    """
+
+    def __init__(self):
+        """
+        Create an :class:`AcceptLanguageNoHeader` instance.
+        """
+        self._header_value = None
+        self._parsed = None
+        self._parsed_nonzero = None
+
+    @property
+    def header_value(self):
+        """
+        (``str``) The header value.
+
+        As there is no header in the request, this is ``None``.
+        """
+        return self._header_value
+
+    @property
+    def parsed(self):
+        """
+        (``list``) Parsed form of the header.
+
+        As there is no header in the request, this is ``None``.
+        """
+        return self._parsed
+
+    def __add__(self, other):
+        """
+        Add to header, creating a new header object.
+
+        `other` can be:
+
+        * a ``str``
+        * a ``dict``, with language ranges as keys and qvalues as values
+        * a ``tuple`` or ``list``, of language range ``str``s or of ``tuple``
+          or ``list`` (language range, qvalue) pairs (``str``s and pairs can be
+          mixed within the ``tuple`` or ``list``)
+        * any object that returns a value for `__str__`
+        * an :class:`AcceptLanguageValidHeader`,
+          :class:`AcceptLanguageNoHeader`, or
+          :class:`AcceptLanguageInvalidHeader` instance
+
+        If neither operands are empty (header value ``''``) or
+        ``None``/:class:`AcceptLanguageNoHeader`, the two header values are
+        joined with ``', '``.
+        """
+        if isinstance(other, AcceptLanguageNoHeader):
+            return self.__class__()
+
+        if isinstance(other, AcceptLanguageValidHeader):
+            return create_accept_language_header(
+                header_value=other.header_value
+            )
+
+        if isinstance(other, AcceptLanguageInvalidHeader):
+            return AcceptLanguageInvalidHeader(header_value=other.header_value)
```

Comment:
Ah sorry, I'd missed that comment somehow. Ignoring invalid headers makes sense to me, especially with your scenario of `request.accept_language += 'en-US;q=0.8'`. I think it depends on what we want addition to mean: my implementation in the PR is adding two header values, whatever they are, and deciding afterwards from the result what class of header it is. Your suggestion is saying, before the addition, that an invalid header is not a thing that should exist; and if we must add an invalid header, then we'll ignore it rather than raise. So the current implementation in the PR is adding them as strings, then reacting to the result; your suggestion would be adding them as individual *headers*, where an invalid header should be ignored. Both make sense in their own way, and it depends on which meaning you want `__add__`/`__radd__` to represent.

If adding an invalid header (as an object) is to be ignored, that means adding an invalid header *value* (as a str, dict, etc.) should also be ignored, right?